python/ray/_private/worker.py

```diff
@@ -2630,50 +2638,62 @@ def put(
 @PublicAPI
 @client_mode_hook
 def wait(
-    object_refs: List["ray.ObjectRef"],
+    ray_waitables: List[RayWaitable[Any]],
     *,
     num_returns: int = 1,
     timeout: Optional[float] = None,
     fetch_local: bool = True,
-) -> Tuple[List["ray.ObjectRef"], List["ray.ObjectRef"]]:
+) -> Tuple[List[RayWaitable[Any]], List[RayWaitable[Any]]]:
```

Comment:
better use a generic type T instead of Any? So we can get stricter type info from type inference.

Reply:
Isn't this technically wrong? My impression is T means all the Waitables will have the same type underlying data, which is not true for this APi. But I am also not very familiar with python typing.

Reply:
addressed

Reply:
So I played with a little. Looks like having T requires us to have various overload because we can have mixture of types. I will leave it to the curernt style for now

Reply:
Let me try and see if it works with pypi

Reply:
Looks like it works. I will use T